lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -372,21 +413,18 @@ export default class Recipe extends RecipeModule {
                 };
             }
             if (shouldDoAccountLinking.shouldRequireVerification) {
-                if (recipeId === "emailpassword" || recipeId === "thirdparty") {
-                    let querier2 = Querier.getNewInstanceOrThrowError("emailverification");
-                    let response = await querier2.sendGetRequest(new NormalisedURLPath("/recipe/user/email/verify"), {
-                        userId: recipeUser.user.id,
-                        email: recipeUser.user.email,
-                    });
-                    if (!response.isVerified) {
-                        return {
-                            createRecipeUser: false,
-                            accountsLinked: false,
-                            reason: "EXISTING_ACCOUNT_NEEDS_TO_BE_VERIFIED_ERROR",
-                        };
-                    }
+                if (!user.loginMethods[0].verified) {
```

Comment:
why just checking `[0]`th element?

Reply:
Cause if the code reaches this part, it means the user is not a primary user and will only have one login method in the array. This is part of an if block that starts on line 378